core/src/main/java/cucumber/runtime/formatter/TimelineFormatter.java

```diff
@@ -0,0 +1,265 @@
+import gherkin.deps.com.google.gson.annotations.SerializedName;
+import gherkin.pickles.PickleTag;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.net.URL;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class TimelineFormatter implements Formatter {
+
+    private static final Comparator<TestData> TEST_DATA_COMPARATOR = new Comparator<TestData>() {
+        @Override
+        public int compare(final TestData o1, final TestData o2) {
+            return o1.id.compareTo(o2.id);
+        }
+    };
+    private static final String CONTENT_TEMPLATE = "%s<br/>%s";
+
+    //TODO: if accepted then should move resources out into own project as per HTML report
+    private static final String[] TEXT_ASSETS = new String[]{
+        "/cucumber/formatter/timeline/index.html",
+        "/cucumber/formatter/timeline/formatter.js",
+        "/cucumber/formatter/timeline/report.css",
+        "/cucumber/formatter/timeline/jquery-3.3.1.min.js",
+        "/cucumber/formatter/timeline/vis.min.css",
+        "/cucumber/formatter/timeline/vis.min.js",
+        "/cucumber/formatter/timeline/vis.override.css",
+        "/cucumber/formatter/timeline/chosen.jquery.min.js",
+        "/cucumber/formatter/timeline/chosen.min.css",
+        "/cucumber/formatter/timeline/chosen.override.css",
+        "/cucumber/formatter/timeline/chosen-sprite.png"
+    };
+
+    private final EventHandler<TestSourceRead> testSourceReadHandler = new EventHandler<TestSourceRead>() {
+        @Override
+        public void receive(TestSourceRead event) {
+            testSources.addTestSourceReadEvent(event.uri, event);
+        }
+    };
+    private final EventHandler<TestCaseStarted> caseStartedHandler = new EventHandler<TestCaseStarted>() {
+        @Override
+        public void receive(TestCaseStarted event) {
+            handleTestCaseStarted(event);
+        }
+    };
+    private final EventHandler<TestCaseFinished> caseFinishedHandler = new EventHandler<TestCaseFinished>() {
+        @Override
+        public void receive(final TestCaseFinished event) {
+            handleTestCaseFinished(event);
+        }
+    };
+    private final EventHandler<TestRunFinished> runFinishedHandler = new EventHandler<TestRunFinished>() {
+        @Override
+        public void receive(final TestRunFinished event) {
+            finishReport(event);
+        }
+    };
+
+    private final TestSourcesModel testSources = new TestSourcesModel();
+    private final List<TestData> allTests = new LinkedList<TestData>();
+    private final Map<Long, GroupData> allGroups = new ConcurrentHashMap<Long, GroupData>();
```

Comment:
Will no longer need to be concurrent

Reply:
Totally forgot to look at this.

Reply:
I’m happy to change, test and verify this formatter

Reply:
Please go ahead. I've changed the `Formatter` interface to `ConcurrentEventListener` and `Plugin`.  I would like to eventually deprecate the `Formatter` interface but that will happen on another PR.

Reply:
I’ll do this tomorrow on the train.  Currently waiting to start an exam